src/sphinxql.y

```diff
@@ -807,6 +807,26 @@ filter_item:
 			if ( !pParser->AddStringFilter ( $1, $3, true ) )
 				YYERROR;
 		}
+	| expr_ident '>' TOK_QUOTED_STRING
+		{
+			if ( !pParser->AddStringCmpFilter ( $1, $3, EStrCmpDir::GT ) )
+				YYERROR;
+		}
+	| expr_ident '<' TOK_QUOTED_STRING
+		{
+			if ( !pParser->AddStringCmpFilter ( $1, $3, EStrCmpDir::LT ) )
+				YYERROR;
+		}
+	| expr_ident TOK_GTE TOK_QUOTED_STRING
+		{
+			if ( !pParser->AddStringCmpFilter ( $1, $3, EStrCmpDir::GTE ) )
+				YYERROR;
+		}
+	| expr_ident TOK_LTE TOK_QUOTED_STRING
+		{
+			if ( !pParser->AddStringCmpFilter ( $1, $3, EStrCmpDir::LTE ) )
+				YYERROR;
+		}
```

Comment:
Ah, part of this was a consequence of the first iteration of this PR, where I rewrote `FilterString_c` to handle all cases. But this was too invasive of a change, so the logic was split out to `FilterStringCmp_c`, leading to the `EQ` and `NE` cases going unused. Very good point about just having fewer cases. I think the enum can just be `{GT, LT}`, and as you point out `m_bExclude` can give `{LTE, GTE}`. I think `EQ` and `NE` can still be handled by `FilterString_c`.

Reply:
@klirichek You suggested having `EQ` in the enum, but what do you think if we also leave `EQ` out (because `FilterString_c` can handle EQ+NE)?

Reply:
Oh, I see. We need `EQ` because of the default case in the filter settings. But perhaps it's more clear to still remove the `EQ` case and instead name the default case `UNSET`?